refactor(linter_codegen): improve code style for collecting nodes

camchenry opened the pull request:
Almost pure refactor. There are some slight styling changes in the rule runner impls file, but other than that, this is just some clean up.

- Make the rule runner template string a multiline string, so it is easier to read and edit
- Move functions into the if-else detector so we don't need to pass around mutable sets of strings
- Replaced BTreeSet with a `NodeTypeSet` that abstracts over the set manipulation somewhat and offers more utilities specific to the linter codegen

camchenry commented:
* **#13637** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13637?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#13636** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13636?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13636?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=13636).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camchenry commented on tasks/linter_codegen/src/main.rs:
```diff
@@ -35,32 +35,35 @@ pub fn generate_rule_runner_impls() -> io::Result<()> {
         write!(
             out,
-            "impl RuleRunner for crate::rules::{plugin_module}::{rule_module}::{rule_struct} {{\n    const NODE_TYPES: &AstTypesBitset = &{node_types_init};\n    const ANY_NODE_TYPE: bool = {any_node_type};\n\n}}\n\n",
+            r"
+impl RuleRunner for crate::rules::{plugin_module}::{rule_module}::{rule_struct} {{
+    const NODE_TYPES: &AstTypesBitset = &{node_types_init};
+    const ANY_NODE_TYPE: bool = {any_node_type};
+}}
+            ",
```
True, but not important, because we `rustfmt` this string in the end anyway.

camc314 commented:
### Merge activity

* **Sep 10, 3:55 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
refactor(linter_codegen): improve code style for collecting nodes (#13636)

Almost pure refactor. There are some slight styling changes in the rule runner impls file, but other than that, this is just some clean up.

- Make the rule runner template string a multiline string, so it is easier to read and edit
- Move functions into the if-else detector so we don't need to pass around mutable sets of strings
- Replaced BTreeSet with a `NodeTypeSet` that abstracts over the set manipulation somewhat and offers more utilities specific to the linter codegen

Co-authored-by: camc314 <18101008+camc314@users.noreply.github.com>
